### ISSUE TITLE
pc files with multiple libraries and include paths

### ISSUE THREAD

**Issue by tylerjw** [cps-org/cps-config#64]
I will eventually migrate my work of writing cps files for ubuntu into some integration tests here.

Here is a cps file I wrote from the below pc file. https://github.com/tylerjw/ubuntu-jammy-cps/blob/main/lib/cps/nss.cps
```
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib/x86_64-linux-gnu
includedir=${prefix}/include/nss

Name: NSS
Description: Mozilla Network Security Services
Version: 3.68.2
Requires: nspr
Libs: -L${libdir} -lnss3 -lnssutil3 -lsmime3 -lssl3
Cflags: -I${includedir}
```

Here is the output of scanning that file using cps-config. You'll note that the include directory is specified every time.
```
./build/cps-config --print-errors --prefix-variable /usr --cflags --libs nss
-I/usr/include/nss -I/usr/include/nss -I/usr/include/nss -I/usr/include/nss -l/usr/lib/x86_64-linux-gnu/libnss3.so -l/usr/lib/x86_64-linux-gnu/libnssutil3.so -l/usr/lib/x86_64-linux-gnu/libsmime3.so -l/usr/lib/x86_64-linux-gnu/libssl3.so
```

Should I specify include paths only in the top-level library and make the other components dependencies of that top-level one, or should I do something else?

**Comment by tylerjw** [comment]
Here is the cps file with them split as components:
```json
{
    "name": "NSS",
    "cps_version": "0.10.0",
    "version": "3.68.2",
    "description": "Mozilla Network Security Services",
    "default_components": ["nss3", "nssutil3", "smime3", "ssl3"],
    "components": {
        "nss3": {
            "type": "dylib",
            "includes": {"c": ["@prefix@/include/nss"]},
            "location": "@prefix@/lib/x86_64-linux-gnu/libnss3.so",
        },
        "nssutil3": {
            "type": "dylib",
            "includes": {"c": ["@prefix@/include/nss"]},
            "location": "@prefix@/lib/x86_64-linux-gnu/libnssutil3.so",
        },
        "smime3": {
            "type": "dylib",
            "includes": {"c": ["@prefix@/include/nss"]},
            "location": "@prefix@/lib/x86_64-linux-gnu/libsmime3.so",
        },
        "ssl3": {
            "type": "dylib",
            "includes": {"c": ["@prefix@/include/nss"]},
            "location": "@prefix@/lib/x86_64-linux-gnu/libssl3.so",
        },
    }
}
```

Here is the file with them merged into a single component using link-libraries:
```json
{
    "name": "NSS",
    "cps_version": "0.10.0",
    "version": "3.68.2",
    "description": "Mozilla Network Security Services",
    "default_components": ["nss3"],
    "components": {
        "nss3": {
            "type": "dylib",
            "includes": {"c": ["@prefix@/include/nss"]},
            "location": "@prefix@/lib/x86_64-linux-gnu/libnss3.so",
            "link-libraries": [
                "@prefix@/lib/x86_64-linux-gnu/libnssutil3.so",
                "@prefix@/lib/x86_64-linux-gnu/libsmime3.so",
                "@prefix@/lib/x86_64-linux-gnu/libssl3.so",
            ],
        },
    }
}
```

Here is a version that doesn't have a location (not standards compliant as the standard requires a location for dylib, maybe it shouldn't?)
```json
{
    "name": "NSS",
    "cps_version": "0.10.0",
    "version": "3.68.2",
    "description": "Mozilla Network Security Services",
    "default_components": ["nss3"],
    "components": {
        "nss3": {
            "type": "dylib",
            "includes": {"c": ["@prefix@/include/nss"]},
            "link-libraries": [
                "@prefix@/lib/x86_64-linux-gnu/libnss3.so",
                "@prefix@/lib/x86_64-linux-gnu/libnssutil3.so",
                "@prefix@/lib/x86_64-linux-gnu/libsmime3.so",
                "@prefix@/lib/x86_64-linux-gnu/libssl3.so",
            ],
        },
    }
}
```

**Comment by tylerjw** [comment]
Closing for now, the below file is what we came to on slack that works well and seems to be what we'd like to encourage:

```json
{
    "name": "NSPR",
    "cps_version": "0.10.0",
    "version": "4.32.0",
    "description": "The Netscape Portable Runtime",
    "default_components": ["nspr4"],
    "components": {
        "nspr4": {
            "type": "archive",
            "includes": {"*": ["@prefix@/include/nspr"]},
            "location": "@prefix@/lib/x86_64-linux-gnu/libnspr4.a",
            "requires": [":plc4", ":plds4"],
        },
        "plc4": {
            "type": "archive",
            "location": "@prefix@/lib/x86_64-linux-gnu/libplc4.a",
        },
        "plds4": {
            "type": "archive",
            "location": "@prefix@/lib/x86_64-linux-gnu/libplds4.a",
        }
    }
}
```
